internal/agent/app/agent.go

```diff
@@ -135,6 +136,32 @@ func (mem *memStorage) makeAndDoRequest(metric Metrics, serverURL string) error
 		mem.logger.Error("can't send metric to the server",
 			zap.Error(err),
 			zap.String("request body: ", string(requestBody)))
+		count := 0
+		for {
+			time.Sleep(time.Second)
+			count++
+			fmt.Println("this ", count)
+			if count == 1 || count == 4 || count == 9 {
+				res, err = http.DefaultClient.Do(req)
+				if err != nil {
```

Comment:
```suggestion
				if err != nil &&count ==9 {
```
можно так записывать. Если первое условие не верно, то дальше сравнение не продолжится